services/web/server/src/simcore_service_webserver/login/registration.py

```diff
@@ -0,0 +1,62 @@
+""" Core functionality and tools for user's registration
+
+
+"""
+# TODO: Move handlers.check_registration  and other utils related with registration here
+import json
+
+from yarl import URL
+
+from ..db_models import ConfirmationAction
+from .storage import AsyncpgStorage
+from .utils import get_expiration_date
+
+
+async def create_invitation(host, guest, db:AsyncpgStorage):
+    """ Creates an invitation token for a guest to register in the platform
+
+        Creates and injects an invitation token in the confirmation table associated
+        to the host user
+
+    :param host: valid user that creates the invitation
+    :type host: Dict-like
+    :param guest: some description of the guest, e.g. email, name or a json
+    """
+    confirmation = await db.create_confirmation(
+        user=host,
+        action=ConfirmationAction.INVITATION.name,
+        data= json.dumps({
+            "created_by": host['email'],
+            "guest": guest
+        })
+    )
+    return confirmation
+
+
+def get_confirmation_info(confirmation):
+    info = confirmation
+
+    # data
+    try:
+        info['data'] = json.loads(confirmation['data'])
```

Comment:
what is the purpose of this? why would json loads fail?

Reply:
The column "data" is a defined in the db model as a string for the moment. Therefore the format that is injected is NOT validated and therefore not guaranteed